extensions-core/druid-lookups/src/main/java/io/druid/server/lookup/PollingLookup.java

```diff
@@ -0,0 +1,259 @@
+ *  or more contributor license agreements. See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership. Metamarkets licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License. You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ * /
+ */
+
+package io.druid.server.lookup;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.ListeningScheduledExecutorService;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import com.metamx.common.ISE;
+import com.metamx.common.StringUtils;
+import com.metamx.common.logger.Logger;
+import io.druid.query.lookup.LookupExtractor;
+import io.druid.server.lookup.cache.polling.OnHeapPollingCache;
+import io.druid.server.lookup.cache.polling.PollingCache;
+import io.druid.server.lookup.cache.polling.PollingCacheFactory;
+import org.joda.time.Period;
+
+import javax.validation.constraints.NotNull;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class PollingLookup extends LookupExtractor
+{
+  private static final Logger LOGGER = new Logger(PollingLookup.class);
+  private static final byte CACHE_ID = 0xa;
+
+  private final Period pollPeriod;
+
+  private final DataFetcher dataFetcher;
+
+  private final PollingCacheFactory cacheFactory;
+
+  private final AtomicReference<CacheRefKeeper> refOfCacheKeeper = new AtomicReference<>();
+
+  private final ListeningScheduledExecutorService scheduledExecutorService = MoreExecutors.listeningDecorator(
+      Executors.newScheduledThreadPool(
+          1,
+          new ThreadFactoryBuilder()
+              .setDaemon(true)
+              .setNameFormat("PollingLookup-%d")
+              .setPriority(Thread.MIN_PRIORITY)
+              .build()
+      )
+  );
+
+  private final AtomicBoolean isOpen = new AtomicBoolean(false);
+
+  private final ListenableFuture<?> pollFuture;
+
+  public PollingLookup(
+      Period pollPeriod,
+      DataFetcher dataFetcher,
+      PollingCacheFactory cacheFactory
+  )
+  {
+    synchronized (isOpen) {
+      this.pollPeriod = pollPeriod == null ? Period.ZERO : pollPeriod;
```

Comment:
why pollPeriod is not "long" but "Period" given that all the access on it converts it to milliseconds ?